lib/src/sql/statements/live.rs

```diff
@@ -66,7 +80,19 @@ impl LiveStatement {
 			}
 		};
 		// Return the query id
-		Ok(self.id.clone().into())
+		Ok(Value::from(self.id.clone()))
+	}
+
+	pub(crate) fn archive(self, node_id: uuid::Uuid) -> LiveStatement {
+		LiveStatement {
+			id: self.id,
+			node: self.node,
+			expr: self.expr,
+			what: self.what,
+			cond: self.cond,
+			fetch: self.fetch,
+			archived: Some(node_id),
+		}
```

Comment:
This works but could be made more concise.
```rust
fn archive(mut self, ...) -> Self
    self.archived = Some(node_id);
    self
}
```

Reply:
I was trying to avoid mutation, but actually you are right. TIL mutation declaration is a cool feature of rust haha :D